kafka/adminapi.go

```diff
@@ -3516,6 +3592,74 @@ func (a *AdminClient) DeleteRecords(ctx context.Context,
 	return result, nil
 }
 
+// ElectLeaders performs Preferred or Unclean Elections for the specified topic Partitions.
+//
+// Parameters:
+//   - `ctx` - context with the maximum amount of time to block, or nil for
+//     indefinite.
+//   - `electLeaderRequest` - ElectLeadersRequest containing the election type
+//     and the partitions to elect leaders for.
+//   - `options` - ElectLeadersAdminOption options.
+//
+// Returns ElectLeadersResult, which contains a slice of TopicPartitions containing the partitions for which the leader election was performed.
+// If we are passing partitions as nil, the broker will perform leader elections for all partitions,
+// but the results will only contain partitions for which there was an election or resulted in an error.
+// Individual TopicPartitions inside the ElectLeadersResult should be checked for errors.
+// Additionally, an error that is not nil for client-level errors is returned.
+func (a *AdminClient) ElectLeaders(ctx context.Context, electLeaderRequest ElectLeadersRequest, options ...ElectLeadersAdminOption) (result ElectLeadersResult, err error) {
+
+	err = a.verifyClient()
+	if err != nil {
+		return result, err
+	}
+
+	cTopicPartitions := newCPartsFromTopicPartitions(electLeaderRequest.partitions)
+	defer C.rd_kafka_topic_partition_list_destroy(cTopicPartitions)
+
+	cElectLeadersRequest := C.rd_kafka_ElectLeaders_new(C.rd_kafka_ElectionType_t(electLeaderRequest.electionType), cTopicPartitions)
+	defer C.rd_kafka_ElectLeaders_destroy(cElectLeadersRequest)
+
+	// Convert Go AdminOptions (if any) to C AdminOptions.
+	genericOptions := make([]AdminOption, len(options))
+	for i := range options {
+		genericOptions[i] = options[i]
+	}
+	cOptions, err := adminOptionsSetup(
+		a.handle, C.RD_KAFKA_ADMIN_OP_ELECTLEADERS, genericOptions)
+	if err != nil {
+		return result, err
+	}
+	defer C.rd_kafka_AdminOptions_destroy(cOptions)
+
+	// Create temporary queue for async operation.
+	cQueue := C.rd_kafka_queue_new(a.handle.rk)
+	defer C.rd_kafka_queue_destroy(cQueue)
+
+	// Call rd_kafka_ElectLeader (asynchronous).
+	C.rd_kafka_ElectLeaders(
+		a.handle.rk,
+		cElectLeadersRequest,
+		cOptions,
+		cQueue)
+
+	// Wait for result, error or context timeout.
+	rkev, err := a.waitResult(
+		ctx, cQueue, C.RD_KAFKA_EVENT_ELECTLEADERS_RESULT)
+	if err != nil {
+		return result, err
+	}
+	defer C.rd_kafka_event_destroy(rkev)
+
+	cResEvent := C.rd_kafka_event_ElectLeaders_result(rkev)
+	cRes := C.rd_kafka_ElectLeaders_result(cResEvent)
+
+	var cResponseSize C.size_t
+	cResultPartitions := C.rd_kafka_ElectLeadersResult_partitions(cRes, &cResponseSize)
+	result.topicPartitions = newTopicPartitionsFromCTopicPartitionResult(cResultPartitions, cResponseSize)
```

Comment:
After librdkafka PR for removing the duplicate result it becomes:
```suggestion
	cRes := C.rd_kafka_event_ElectLeaders_result(rkev)

	var cResponseSize C.size_t
	cResultPartitions := C.rd_kafka_ElectLeaders_result_partitions(cRes, &cResponseSize)
	result.TopicPartitions = newTopicPartitionsFromCTopicPartitionResult(cResultPartitions, cResponseSize)

```